packages/blocks/src/api/factory.js

```diff
@@ -47,7 +47,18 @@ export function createBlock( name, blockAttributes = {}, innerBlocks = [] ) {
 		if ( undefined !== value ) {
 			result[ key ] = value;
 		} else if ( schema.hasOwnProperty( 'default' ) ) {
-			result[ key ] = schema.default;
+			let defaultValue = schema.default;
+
+			// If the type of the attribute's default is a reference type,
+			// make a copy so that implementors cannot easily mutate the
+			// block type's default attributes.
+			if ( Array.isArray( defaultValue ) ) {
```

Comment:
Not sure if we should consider deep cloning?

Reply:
Maybe we could use `_.clone`?

https://lodash.com/docs/4.17.10#clone

Reply:
Nice! I was sure I searched for a `clone` in the lodash docs and couldn't find anything, but there it is, clear as day.